tests/cstor/gtest/test_zrepl_prot.cc

```diff
@@ -296,6 +296,42 @@ static void get_zvol_status(std::string zvol_name, uint64_t &ioseq, int control_
 	EXPECT_EQ(status.rebuild_status, rebuild_status);
 }
 
+static void wait_for_replica_status(std::string zvol_name, uint64_t &ioseq,
+    int control_fd, int state, struct zrepl_status_ack &status,
+    int version = REPLICA_VERSION) {
+	zvol_io_hdr_t hdr_in, hdr_out = {0};
+	int rc, max_retry_count=10, retry_count = 0;
+
+	while (retry_count < max_retry_count) {
+		hdr_out.version = version;
+		hdr_out.opcode = ZVOL_OPCODE_REPLICA_STATUS;
+		hdr_out.status = ZVOL_OP_STATUS_OK;
+		hdr_out.io_seq = ++ioseq;
+		hdr_out.len = zvol_name.length() + 1;
+		rc = write(control_fd, &hdr_out, sizeof (hdr_out));
+		ASSERT_ERRNO("write", rc >= 0);
+		ASSERT_EQ(rc, sizeof (hdr_out));
+		rc = write(control_fd, zvol_name.c_str(), hdr_out.len);
+		ASSERT_ERRNO("write", rc >= 0);
+		ASSERT_EQ(rc, hdr_out.len);
+		rc = read(control_fd, &hdr_in, sizeof (hdr_in));
+		ASSERT_ERRNO("read", rc >= 0);
+		ASSERT_EQ(rc, sizeof (hdr_in));
+		EXPECT_EQ(hdr_in.version, hdr_out.version);
+		EXPECT_EQ(hdr_in.opcode, ZVOL_OPCODE_REPLICA_STATUS);
+		EXPECT_EQ(hdr_in.io_seq, ioseq);
+		EXPECT_EQ(hdr_in.status, ZVOL_OP_STATUS_OK);
+		EXPECT_EQ(hdr_in.len, sizeof (status));
+		rc = read(control_fd, &status, sizeof (status));
+		ASSERT_ERRNO("read", rc >= 0);
+		if (status.state == state) {
```

Comment:
We should fail if we exceeded the max_retry_count. Here we just assume that vol is healthy if we have exceeded the max count.

Reply:
never mind, I see you checking the status again after returning from this function.

Reply:
it better to put that check here in this function only.

Reply:
Yes done